app/User.php

```diff
@@ -374,4 +397,46 @@ public function accessOverrideBy(): BelongsTo
     {
         return $this->belongsTo(self::class, 'access_override_by_id');
     }
+
+    /**
+     * Synchronizes major relationship with a given list of gtAccountEntitlements.
+     *
+     * @param array<string>  $accountEntitlements
+     */
+    public function syncMajorsFromAccountEntitlements(array $accountEntitlements): void
```

Comment:
Why is this in the user model? I feel like this would be better somewhere else - maybe a job?

Reply:
This will be used in multiple places including background jobs and during login.